FIX: Various payment - Missing fields for check transmitter & bank name

aspangaro opened the pull request:
![image](https://user-images.githubusercontent.com/2341395/108311675-b08d4680-71b5-11eb-9007-2e1c11ea6914.png)

eldy commented on htdocs/compta/bank/various_payment/card.php:
```diff
@@ -390,20 +432,28 @@
-	$form->select_types_paiements($paymenttype, "paymenttype");
-	print '</td></tr>';
+	print '<tr><td><span class="fieldrequired">'.$langs->trans('PaymentMode').'</span></td><td>';
+	$form->select_types_paiements($paymenttype, 'paymenttype', '', 2);
```
You changed the value returned by the select by returning code instead of id but i did not found modification of code that used the value posted. 
And code to get the value is 
GETPOST('paymenttype', 'int')
So this seems to introduce a bug. Id must be prefered to code when we have an id for key.

eldy committed:
Merge pull request #16373 from aspangaro/14a20

Code enhanced - Standardize field name #14697

eldy committed:
Merge branch 'develop' into 14a21

eldy committed:
Update card_presend.tpl.php

eldy committed:
Merge pull request #16379 from danielprowald/develop

CLOSE #16378 More E-Mail Contact substitution Values for better salutation

eldy committed:
Update subscription.php

eldy committed:
Merge pull request #16370 from bafbes/abb120367

new:hook doActions for subscriptions with possibility to modify values of variables $datefrom,$dateto,$paymentdate inside hook

frederic34 committed:
Update main.inc.php

BB2A-Anthony committed:
Update functions.lib.php

frederic34 committed:
complete doaction hook

eldy committed:
Merge pull request #16374 from aspangaro/14a21

Code enhanced - Standardize field name #14697

eldy committed:
Fix website editor

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
fix box activity

eldy committed:
Merge pull request #16381 from frederic34/patch-3

fix box activity

defrance committed:
fix Notice error on install

frederic34 committed:
fix propal sqls

eldy committed:
Update inc.php

eldy committed:
Merge pull request #16382 from defrance/patch-164

fix Notice error on install

eldy committed:
Merge pull request #16380 from frederic34/membersubscription

complete doaction hook

eldy committed:
Merge pull request #16384 from frederic34/boxpropale

fix propal sqls

eldy committed:
Merge pull request #16371 from frederic34/patch-2

add log for token reclaimed

eldy committed:
Merge pull request #16357 from bb2a/ADD_SUBSTITUTION_DELIVRY_DATE

Add substitution for delivry date

frederic34 committed:
remove space and code syntax

frederic34 committed:
code syntax

stickler-ci committed:
Fixing style errors.

eldy committed:
Responsive

eldy committed:
Merge branch '13.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into installinc

daraelmin committed:
Fix #16096 #16085 missing CSRF token

eldy committed:
Merge pull request #16385 from frederic34/installinc

remove space and code syntax

frederic34 committed:
fix propal fetch

frederic34 committed:
code syntax

frederic34 committed:
code syntax

frederic34 committed:
code syntax

frederic34 committed:
Update propal.class.php

FHenry committed:
rename column

FHenry committed:
Merge branch 'dev_EventOrganisation_draft' of github.com:FHenry/dolibarr into dev_EventOrganisation_draft

FHenry committed:
Merge branch 'develop' into dev_EventOrganisation_draft

frederic34 committed:
code syntax

frederic34 committed:
css

eldy committed:
Merge pull request #16394 from daraelmin/daraelmin-patch-3

Fix #16096 missing CSRF token and use post instead get

eldy committed:
Merge pull request #16391 from frederic34/assetcodesyntax

code syntax

eldy committed:
Merge pull request #16390 from frederic34/patch-2

fix propal fetch

eldy committed:
Merge pull request #16392 from frederic34/patch-3

css

eldy committed:
Merge pull request #16368 from FHenry/dev_EventOrganisation_draft

NEW : Tab on project for Event Organization #FoundationFunding

frederic34 committed:
add cache on project head

frederic34 committed:
fix typo

eldy committed:
NEW Can set a percentage when creating an invoice from another object

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Code syntax

frederic34 committed:
code syntax api directory

frederic34 committed:
code syntax bom directory

frederic34 committed:
code syntax cashdesk directory

frederic34 committed:
code syntax categorie collab directory

frederic34 committed:
fix fourn card

frederic34 committed:
code syntax comm directory

frederic34 committed:
code syntax comm directory

frederic34 committed:
code syntax comm directory

frederic34 committed:
code syntax comm directory

frederic34 committed:
code syntax comm directory

frederic34 committed:
code syntax comm directory

frederic34 committed:
code syntax comm directory

frederic34 committed:
code syntax commande directory

frederic34 committed:
code syntax compta directory

frederic34 committed:
code syntax contact directory

frederic34 committed:
code syntax contrat directory

frederic34 committed:
code syntax core directory

eldy committed:
Removed useless field

eldy committed:
Fix GETPOST instead of POST

eldy committed:
Code comment

eldy committed:
Var not defined

eldy committed:
Prepare public page for autocreate contact-lead

eldy committed:
WIP Add lead

eldy committed:
Fix json decode

dvdwalker committed:
Merge pull request #1 from Dolibarr/develop

Update to 13.0

dvdwalker committed:
Update product.class.php

fappels committed:
Merge remote-tracking branch 'Dolibarr/develop' into develop_mo_pmp

inoveaconseil committed:
NEW option for Takepos to show the total price without tax

fappels committed:
Fix displayed unit cost of free consumption line

inoveaconseil committed:
Correct stickler

inoveaconseil committed:
Correct stickler

frederic34 committed:
typo

dvdwalker committed:
Fix: Ajax on/off not saving value in DB after updating to version >=12

The issue seams to be that starting in version 12 the "[element]_UPDATE" trigger is called in file core/ajax/objectonoff.php but some properties that are required on $object are not set. Those are (at least for product that is what I use):

$object->id
$object->table_element
$object->fields[$field]

The error occurs on the setValueFrom function in core/class/commonobject.class.php file

josemariagomezroncero committed:
Update ProductCombination.class.php

Product dont update product variation statut when desactivate for sale.

dvdwalker committed:
Fix: Ajax on/off not saving value in DB after updating to version >=12

The issue seams to be that starting in version 12 the "[element]_UPDATE" trigger is called in file core/ajax/objectonoff.php but some properties that are required on $object are not set.

stickler-ci committed:
Fixing style errors.

bafbes committed:
new:Fetch region_code in societe fetch

fappels committed:
Also register movement for 0 produced to update cost.
Use correct movement label for negative consumed.

Commit:
differentiate serial numbers from lot management

Commit:
change hasBatch function

Commit:
add event management to batch products

Commit:
fix for stickler

frederic34 committed:
fix translation

frederic34 committed:
code syntax datapolicy dav debugbar directories

frederic34 committed:
code syntax delivery directory

frederic34 committed:
code syntax don directory

frederic34 committed:
code syntax ecm emailcollector directories

frederic34 committed:
code syntax expedition directory

frederic34 committed:
code syntax expensereport directory

frederic34 committed:
code syntax exports externalsite directory

frederic34 committed:
code syntax fichinter directory

frederic34 committed:
code syntax fourn directory

daraelmin committed:
Fix #16420 undefined  $model_mail_selected_id

Set $model_mail_selected_id to 0 when it is undefined

eldy committed:
Merge pull request #16413 from frederic34/contact_codesyntax

code syntax contact directory

eldy committed:
Merge pull request #16412 from frederic34/compta_codesyntax

code syntax compta directory

eldy committed:
Merge pull request #16411 from frederic34/commande_codesyntax

code syntax commande directory

eldy committed:
Merge pull request #16410 from frederic34/comm_codesyntax

code syntax comm directory

eldy committed:
Merge pull request #16409 from frederic34/patch-3

fix fourn card

eldy committed:
Merge pull request #16408 from frederic34/categ_codesyntax

code syntax categorie collab directory

eldy committed:
Merge pull request #16415 from frederic34/core_code_syntax

code syntax core directory

eldy committed:
Merge pull request #16446 from frederic34/fichinter_code_syntax

code syntax fichinter directory

eldy committed:
Merge pull request #16447 from frederic34/fourn_code_syntax

code syntax fourn directory

eldy committed:
Merge pull request #16445 from frederic34/exports_code_syntax

code syntax exports externalsite directory

eldy committed:
Merge pull request #16444 from frederic34/expensereport_code_syntax

code syntax expensereport directory

eldy committed:
Merge pull request #16443 from frederic34/expedition_code_syntax

code syntax expedition directory

eldy committed:
Merge pull request #16441 from frederic34/don_code_syntax

code syntax don directory

eldy committed:
Update html.formmail.class.php

eldy committed:
Merge pull request #16449 from daraelmin/daraelmin-patch-3

Fix #16420 undefined  $model_mail_selected_id

eldy committed:
Merge pull request #16439 from frederic34/datapolicy_code_syntax

code syntax datapolicy dav debugbar directories

eldy committed:
Merge pull request #16440 from frederic34/delivery_code_syntax

code syntax delivery directory

eldy committed:
Merge pull request #16438 from frederic34/patch-5

fix translation

eldy committed:
Merge pull request #16442 from frederic34/ecm_code_syntax

code syntax ecm emailcollector directories

eldy committed:
Merge pull request #16427 from dvdwalker/ajax-fix

Fix: Option MAIN_DIRECT_STATUS_UPDATE broken. Ajax on/off not saving value in DB after updating to version >=12

eldy committed:
Fix init before

eldy committed:
Look and feel v14

eldy committed:
Fix phpcs

eldy committed:
Merge pull request #16435 from altairisfr/batchevents_pr

add event management to batch products

Commit:
change translation

frederic34 committed:
code syntax zapier directory

frederic34 committed:
code syntax website directory

eldy committed:
NEW: Can create a lot from list
Make code like modulebuilder

eldy committed:
Merge branch '13.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	htdocs/compta/facture/card.php
	htdocs/core/class/html.formmail.class.php
	htdocs/core/lib/product.lib.php
	htdocs/product/stock/productlot_card.php
	test/phpunit/SecurityTest.php

eldy committed:
Show product on lot record tab document

eldy committed:
Merge pull request #16454 from frederic34/website_code_syntax

code syntax website directory

eldy committed:
Merge pull request #16424 from frederic34/patch-4

typo

eldy committed:
Merge pull request #16400 from frederic34/patch-2

fix typo

eldy committed:
Merge pull request #16422 from inovea-conseil/takepos_priceht

NEW option for Takepos

frederic34 committed:
really fix typo

eldy committed:
Merge branch 'develop' into count_project_head

eldy committed:
Merge pull request #16399 from frederic34/count_project_head

add cache on project head

eldy committed:
Merge pull request #16455 from frederic34/patch-3

really fix typo

eldy committed:
Merge pull request #16405 from frederic34/api_codesyntax

code syntax api directory

eldy committed:
Merge pull request #16407 from frederic34/casdesk_code_syntax

code syntax cashdesk directory

eldy committed:
Merge pull request #16414 from frederic34/contract_code_syntax

code syntax contrat directory

eldy committed:
Merge pull request #16406 from frederic34/bom_codesyntax

code syntax bom directory

eldy committed:
Update societe.class.php

frederic34 committed:
code syntax variants webservice directory

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into zapier_code_syntax

stickler-ci committed:
Fixing style errors.

eldy committed:
Update societe.class.php

eldy committed:
Merge pull request #16429 from bafbes/abb120369

Add region code and id into fetch of thirdparty

eldy committed:
Merge branch 'develop' into develop

eldy committed:
Merge pull request #16421 from dvdwalker/develop

Minor Typo on tooltip class prenventing correct format to show

frederic34 committed:
code syntax user directory

eldy committed:
Merge pull request #16453 from frederic34/zapier_code_syntax

code syntax zapier directory

eldy committed:
Merge pull request #16456 from frederic34/variants_code_syntax

code syntax variants webservice directory

eldy committed:
Code comment

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Update ProductCombination.class.php

eldy committed:
Merge pull request #16426 from josemariagomezroncero/patch-9

NEW Set status of all variant when setting status of parent

eldy committed:
Merge pull request #16457 from frederic34/user_code_syntax

code syntax user directory

frederic34 committed:
code syntax ticket directory

frederic34 committed:
code syntax theme directory

eldy committed:
Code comment

eldy committed:
Merge pull request #16458 from frederic34/ticket_code_syntax

code syntax ticket directory

eldy committed:
Merge pull request #16459 from frederic34/theme_code_syntax

code syntax theme directory

frederic34 committed:
code syntax product directory

eldy committed:
Fix removed a bugged section (not used, was in develop mode only)

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Clean code

eldy committed:
Merge pull request #16434 from altairisfr/serials_pr

differentiate serial numbers from lot management

eldy committed:
Trans

frederic34 committed:
code syntax printing directory

frederic34 committed:
code syntax paypal paybox directory

frederic34 committed:
code syntax opensurvey directory

fappels committed:
MoLine origin is bomline not bom

stickler-ci committed:
Fixing style errors.

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into product_code_syntax

frederic34 committed:
Merge remote-tracking branch 'origin/product_code_syntax' into product_code_syntax

frederic34 committed:
code syntax directory

eldy committed:
Merge pull request #16112 from fappels/develop_mo_pmp

Add missing pmp valorisation to MO production. (Issue #16072)

eldy committed:
Merge pull request #16460 from frederic34/product_code_syntax

Product code syntax

daraelmin committed:
Fix #16418 missing title in "internal transfer"

daraelmin committed:
Fix #16418 title in "Transaction list by category"

daraelmin committed:
Fix #16418 missing in "Tags of transactions"

daraelmin committed:
Fix Dolibarr#16418 missing title in "iNew cCash Fence"

frederic34 committed:
code syntax ftp holiday directory

stickler-ci committed:
Fixing style errors.

frederic34 committed:
code syntax hrm imports install intracommreport loan directory

frederic34 committed:
code syntax m... directories

stickler-ci committed:
Fixing style errors.

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into m_code_syntax

frederic34 committed:
Merge remote-tracking branch 'origin/m_code_syntax' into m_code_syntax

frederic34 committed:
code syntax

frederic34 committed:
code syntax project

frederic34 committed:
code syntax public dir

eldy committed:
Fix missing in v13

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch '13.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	htdocs/accountancy/bookkeeping/balancebymonth.php
	htdocs/accountancy/class/accountancyexport.class.php

eldy committed:
Merge pull request #16462 from frederic34/ftpholiday_code_syntax

code syntax ftp holiday directory

eldy committed:
Merge pull request #16463 from frederic34/hrm_code_syntax

code syntax hrm imports install intracommreport loan directory

eldy committed:
Merge pull request #16464 from frederic34/m_code_syntax

code syntax m... directories

eldy committed:
Merge pull request #16461 from daraelmin/daraelmin-patch-4

Daraelmin patch 4

eldy committed:
Merge pull request #16467 from frederic34/project_code_syntax

code syntax project

eldy committed:
Merge branch '13.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

frederic34 committed:
code syntax r... dir

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into public_code_syntax

frederic34 committed:
code syntax s... t..  dir

frederic34 committed:
code syntax eventorganization dir

frederic34 committed:
code syntax admin dir

frederic34 committed:
fix variable name

frederic34 committed:
fix cache

eldy committed:
NEW Add link picto to the stock movement on the detail of production

eldy committed:
Look and feel v14

eldy committed:
Look and feel v14

eldy committed:
Look and feel v14

eldy committed:
Look and feel v14

eldy committed:
Fix field hidden by default (not used)

eldy committed:
Look and feel v14

frederic34 committed:
fix missing include

needed by dol_get_last_hour

frederic34 committed:
add update CommandeFournisseur for api use

eldy committed:
Look and feel v14

eldy committed:
Look and feel v14

eldy committed:
Code change to follow modulebuilder template

eldy committed:
Show 3 years in graph of widget by default.

eldy committed:
Trans

eldy committed:
Look and feel v14

eldy committed:
Merge pull request #16478 from frederic34/patch-5

fix missing include

eldy committed:
Merge pull request #16473 from frederic34/patch-3

fix variable name

eldy committed:
Merge pull request #16472 from frederic34/admin_code_syntax

code syntax admin dir

eldy committed:
Merge pull request #16471 from frederic34/eventorg_code_syntax

code syntax eventorganization dir

eldy committed:
Merge pull request #16479 from frederic34/update_CommandeFournisseur

add update CommandeFournisseur for api use

eldy committed:
Merge pull request #16468 from frederic34/public_code_syntax

code syntax public dir

eldy committed:
Merge branch '13.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	htdocs/commande/card.php
	htdocs/core/boxes/box_factures_fourn.php
	htdocs/core/boxes/box_factures_fourn_imp.php
	htdocs/core/boxes/box_ficheinter.php
	htdocs/core/boxes/box_members.php
	htdocs/core/boxes/box_propales.php
	htdocs/core/class/commonobject.class.php
	htdocs/core/class/html.formmail.class.php
	htdocs/core/lib/functions.lib.php
	htdocs/fourn/class/fournisseur.facture.class.php

eldy committed:
Look and feel v14

eldy committed:
Fix align

frederic34 committed:
fix

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into r_code_syntax

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into s_code_syntax

eldy committed:
NEW Add last date of modification for website pages in list

eldy committed:
Look and feel v14

eldy committed:
FIX Timezone problems

eldy committed:
Merge pull request #16469 from frederic34/r_code_syntax

code syntax r... dir

eldy committed:
Merge pull request #16470 from frederic34/s_code_syntax

code syntax s... t..  dir

aspangaro committed:
FIX: Various payment - Missing fields for check transmitter & bank name

aspangaro committed:
Merge remote-tracking branch 'upstream/develop' into 13p6

aspangaro committed:
Revert "Merge remote-tracking branch 'upstream/develop' into 13p6"

This reverts commit 86aeb21f91f4e4f1bf0e7f46a34323381200ed01, reversing
changes made to 5726850cca3f33a47812a9d7b09da1f7fe9e7f55.

aspangaro committed:
Merge branch '13.0' into 13p6

aspangaro commented:
Problem of merge...
